src/JWT.ts

```diff
@@ -258,17 +249,16 @@ export type GeneralJWSSignature = {
 export function decodeJWT(jwt: string, recurse = true): JWTDecoded {
   if (!jwt) throw new Error('invalid_argument: no JWT passed into decodeJWT')
   try {
-    const signatures: GeneralJWSSignature[] = []
     const jws = decodeJWS(jwt)
     const decodedJwt: JWTDecoded = Object.assign(jws, { payload: JSON.parse(decodeBase64url(jws.payload)) })
+    const iss = decodedJwt.payload.iss
+
     if (decodedJwt.header.cty === 'JWT' && recurse) {
-      signatures.push({
-        protected: decodedJwt.header,
-        signature: decodedJwt.signature as string,
-      })
-      return decodeJWT(decodedJwt.payload.jwt)
+      const innerDecodedJwt = decodeJWT(decodedJwt.payload.jwt)
+
+      if (innerDecodedJwt.payload.iss !== iss) throw new Error(`${JWT_ERROR.INVALID_JWT}: multiple issuers`)
+      return innerDecodedJwt
```

Comment:
this will never be thrown as it gets caught inside the try catch block

Reply:
it will be thrown, and added to the stack trace for the try catch error